### PR TITLE
Theme management command

### LINKED ISSUE #11: ODRC: Toevoegen themas
**Description (why)**
uit PSA: Een beheerder zal in het ODRC thema’s willen kunnen toevoegen, raadplegen, wijzigen en (de)activeren. Thema’s moeten in de data-opslag vastgelegd worden en publicaties moeten hieraan gekoppeld kunnen worden. Deze informatie moet via de API’s ontsloten worden.

**Acceptance criteria (what)**

**Specific details (how)**

**Test plan**

Waardelijst: https://identifier.overheid.nl/tooi/set/scw_toplijst

Note: a theme can be a child of another theme, and the JSON structure above doesn't rule out that a theme is a (direct) child of multiple different other themes

Theme properties:

* `identifier`: URI
* `name`: name of the theme
* `subthema's`: array of nested themes, which may all have their own sub themes (!)
* `publicatieStatus`: actief/niet_actief
* `origin`: enum, `waardelijst` or `zelf_toegevoegd`

**Tasks**

- [x] Define data model for `Thema` and register in the admin. Probably best to use django-treebeard for the tree structure (MPTT)
- [x] Expose via API at `/api/v1/themas`
    - List endpoint is paginated
    - List endpoint has search parameter on `name`
    - List endpoint includes nested themes if relevant, for each item in the list
    - Create/update for origin `zelf_toegevoegd`, waardelijst records may not be mutated
    - publicatieStatus can be updated irrespective of `origin`
    - When creating a theme, you must be able to specify it needs to be a child of another theme (and specify before/after sibling)
- [x] Add management command to populate themes from waardelijst
    - Must produce a fixture, with natural keys
    - Fixture needs to be baked in docker image and automatically loaded on application start (post-migrate signal)

### PR DESCRIPTION
**Pull request by bart-maykin**
Fixes #11 

**Changes**

Added magagement command to store theme waardenlijsten data and turn it into a fixture

### CODE REVIEW

**Comment by bart-maykin** [comment]
I renamed the `waardenlijst_sync.py` file to `information_category_sync.py` to make these sync files more readable. Is this a good decision or should we not split the code up in different files? If we go this route then it might also be an idea to create a dir to store them in a better place.

**Comment by sergei-maertens** [comment]
> I renamed the `waardenlijst_sync.py` file to `information_category_sync.py` to make these sync files more readable. Is this a good decision or should we not split the code up in different files? If we go this route then it might also be an idea to create a dir to store them in a better place.

separate files is great, once we have all three management commands and implementations, we can see if abstractions/refactors/restructuring are in order or not :)

File: src/woo_publications/metadata/theme_sync.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+}
+
+
+class ThemeWaardenlijstError(Exception):
+    def __init__(self, message: str):
+        self.message = message
+        super().__init__(message)
+
+
+def update_theme(file_path: Path):
```
Comment (sergei-maertens) [inline src/woo_publications/metadata/theme_sync.py:29]: `file_path` here is annotated as a `glom.Path`, not a `pathlib.Path` which is causing you type errors.

Removing `from glom import Path` and replacing it with `from pathlib import Path` should fix all issues

Comment (bart-maykin) [reply src/woo_publications/metadata/theme_sync.py:30]: Yes just noticed it myself...

**Comment by bart-maykin** [comment]
> > I renamed the `waardenlijst_sync.py` file to `information_category_sync.py` to make these sync files more readable. Is this a good decision or should we not split the code up in different files? If we go this route then it might also be an idea to create a dir to store them in a better place.
> 
> separate files is great, once we have all three management commands and implementations, we can see if abstractions/refactors/restructuring are in order or not :)

Cool then we are on the same page. I was looking into making the code better because there is a lot of duplicate code. But then I remembered that we are gonna have another command so I left it as it is.

File: src/woo_publications/fixtures/themes.json
Comment (sergei-maertens) [inline src/woo_publications/fixtures/themes.json:1]: self-reminder: ensure this fixture is loaded on container startup

File: src/woo_publications/metadata/theme_sync.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    ]
+
+    root_themes_data = [theme["data"] for theme in waardenlijst if not theme["parents"]]
+    sub_themes = [theme for theme in waardenlijst if theme["parents"]]
+
+    # create/update root themes
+    for theme in root_themes_data:
+        theme_object = Theme.objects.filter(identifier=theme["identifier"])
+        if theme_object.exists():
+            theme_object.update(**theme)
```
Comment (sergei-maertens) [inline src/woo_publications/metadata/theme_sync.py:60]: can you address the missing test coverage?

File: src/woo_publications/metadata/theme_sync.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        else:
+            Theme.add_root(**theme)
+
+    # create/update sub themes
+    for theme in sub_themes:
+        theme_object = Theme.objects.filter(identifier=theme["data"]["identifier"])
+        for parent in theme["parents"]:
+            if theme_object.exists():
+                theme_object.update(**theme["data"])
```
Comment (sergei-maertens) [inline src/woo_publications/metadata/theme_sync.py:69]: this wouldn't take into account if the parent changes, would it?

**Review by sergei-maertens — CHANGES_REQUESTED** [review]

**Review by sergei-maertens — APPROVED** [review]
Nice one!

**Pull request merged**